lib/hash.cc

```diff
@@ -144,42 +125,42 @@ hash_join(hash_table * hid, hash_link * lnk)
  *  returns NULL.
  */
 hash_link *
-hash_lookup(hash_table * hid, const void *k)
+hash_table::hash_lookup(const void *k)
 {
     int b;
     PROF_start(hash_lookup);
     assert(k != NULL);
-    b = hid->hash(k, hid->size);
-    for (hash_link *walker = hid->buckets[b]; walker != NULL; walker = walker->next) {
-        if ((hid->cmp) (k, walker->key) == 0) {
+    b = hash(k, size);
+    for (hash_link *walker = buckets[b]; walker != NULL; walker = walker->next) {
+        if (cmp(k, walker->key) == 0) {
             PROF_stop(hash_lookup);
             return (walker);
         }
         assert(walker != walker->next);
     }
     PROF_stop(hash_lookup);
-    return NULL;
+    return nullptr;
```

Comment:
No need to modify NULL here. It is the only change to this line and can be automated by the Squid-wide enforcer later.